gnat2goto/ireps/ireps_generator.py

```diff
@@ -205,2067 +205,2071 @@ def mk_prefixed_lines(prefix, lines, join=""):
+        # schema -> friendly_name -> (str|int|bool|sloc, index, irep|list|trivial)
+
+        for sn in top_sorted_sn:
+            op_counts[sn] = {"int"  : 0,
+                            "bool" : 0}
+            layout[sn] = {}
+
+            for setter_name, data in sub_setters.iteritems():
+                assert len(data) == 1
+                assert "value" in data or "list" in data
+                typ = "list" if "list" in data else "irep"
+                for kind, variants in data.iteritems():
+                    if sn in variants:
+                        layout[sn][setter_name] = ("int", op_counts[sn]["int"], typ)
+                        op_counts[sn]["int"] += 1
 
-        # cnst ::= schema -> id|namedSub|comment -> {name: value}
-        for kind in const.get(sn, {}):
-            for const_name, const_value in const[sn][kind].iteritems():
-                tmp = "constant %s: %s" % (ada_casing(const_name), const_value)
-                if kind != "id":
-                    tmp += " (%s)" % kind
-                write(s, "--  %s" % tmp)
+            for setter_name, setter_kinds in named_setters.iteritems():
+                for kind in setter_kinds:
+                    if sn in setter_kinds[kind]:
+                        is_comment, typ, _ = setter_kinds[kind][sn]
+                        if kind in ("irep", "list") or typ == "integer":
+                            l_typ = "trivial" if typ == "integer" else kind
+                            layout[sn][setter_name] = ("int",
+                                                    op_counts[sn]["int"],
+                                                    l_typ)
+                            op_counts[sn]["int"] += 1
+                        elif typ == "string":
+                            layout[sn][setter_name] = ("str",
+                                                    op_counts[sn]["int"],
+                                                    "trivial")
+                            op_counts[sn]["int"] += 1
+                        elif typ == "bool":
+                            layout[sn][setter_name] = ("bool",
+                                                    op_counts[sn]["bool"],
+                                                    "trivial")
+                            op_counts[sn]["bool"] += 1
+                        elif typ == "gnat:sloc":
+                            layout[sn][setter_name] = ("sloc",
+                                                    op_counts[sn]["int"],
+                                                    "trivial")
+                            op_counts[sn]["int"] += 1
+                        else:
+                            print sn, setter_name, kind, typ
+                            assert False
+
+        MAX_INTS  = max(x["int"] for x in op_counts.itervalues())
+        MAX_BOOLS = max(x["bool"] for x in op_counts.itervalues())
+
+        if optimize:
+            layout, MAX_INTS, MAX_BOOLS = self.optimize_layout(layout,
```

Comment:
This seems to have been broken in the previous version? It looks like this used to be an empty `if`, which isn't actually valid in Python.

Reply:
I just checked it with meld locally. It seems it's a problem with github diff tool or maybe the "ignore whitespace" option. It was working before, it is also working now.